### PR TITLE
Do dependency checking via a validation action

### PR DESCRIPTION
**Pull request by jadenPete**
You can read more about validation actions here:
https://bazel.build/extending/rules#validation_actions

This should enable dependency checking to be performed concurrently with `phase_singlejar`.

### CODE REVIEW

**Comment by DavidANeil** [comment]
Have you tested that `--norun_validations` correctly skips all of the validations? (i.e. all of the outputs are correctly segregated)

**Comment by DavidANeil** [comment]
For when we pull this into our main repo you will want to experiment with `--experimental_use_validation_aspect` to see if it gives better scheduling.

**Comment by jadenPete** [comment]
> Have you tested that `--norun_validations` correctly skips all of the validations? (i.e. all of the outputs are correctly segregated)

Good idea. I just tried building `//dependencies/indirect:used_deps_off_c` from within `tests` without `--norun_validations`, and a `buildozer` message appeared. I also built it with `--norun_validations`, and the message didn't appear, so it looks like we're good!

**Comment by jadenPete** [comment]
> For when we pull this into our main repo you will want to experiment with `--experimental_use_validation_aspect` to see if it gives better scheduling.

Will do.

File: rules/private/phases/phase_zinc_depscheck.bzl [hunk 1]
```diff
@@ -61,21 +60,16 @@ def phase_zinc_depscheck(ctx, g):
...
         )
-        deps_checks[name] = deps_check
 
-    outputs = []
-    if deps_configuration.direct == "error":
-        outputs.append(deps_checks["direct"])
-    if deps_configuration.used == "error":
-        outputs.append(deps_checks["used"])
+        if getattr(deps_configuration, name) == "error":
+            outputs.append(deps_check)
```
Comment (jjudd) [inline rules/private/phases/phase_zinc_depscheck.bzl:65]: If you removed `deps_check` from line 24, then where is `deps_check` coming from?

**Review by jjudd — APPROVED** [review]

File: rules/private/phases/phase_zinc_depscheck.bzl (hunk 1 above)
Comment (jadenPete) [reply rules/private/phases/phase_zinc_depscheck.bzl:65]: It should be coming from line 28.

Comment (jjudd) [reply rules/private/phases/phase_zinc_depscheck.bzl:65]: Totally missed that. My bad.

**Pull request merged**